2019Labs/RHELSecurityLab/documentation/lab3_NBDE_Cockpit.adoc

```diff
@@ -0,0 +1,137 @@
+
+== Introduction
+Network-Bound Disk Encryption (NBDE) allows the user to encrypt root volumes of hard drives on physical and virtual machines without requiring to manually enter a password when systems are restarted.  Red Hat Enterprise Linux implements NBDE with a central server named Tang, and and client framework named Clevis.
+
+Tang is used to bind data to a secure network.  It is stateless and does not require TLS or authentication.  Tang does not interact with client keys, so it never obtains identifying information from the client.  Clevis provides automated unlocking of LUKS volumes.
+
+Clevis and Tang are generic client and server components that provide network bound encryption. In Red Hat Enterprise Linux 7, they are used in conjunction with LUKS to encrypt and decrypt root and non-root storage volumes to accomplish Network Bound Disk Encryption.
+
+== Lab 3.1 Installing the Tang server
+. Navigate to the Lab information page from *Lab 0 Setup steps*. This page has your environment's power control and consoles. Click on the console for *NBDE1*. Login to the NBDE1 server as *root* using the password *r3dh4t1!*.
+
++
+image:images/lab3-console.png[200,200]
+
+. Then, install the tang packages
++
+[source, text]
+[root@nbde1 ~]# yum install tang
++
+
+NOTE:
+If you wanted to copy and paste text to send to the console, you could also press the *T* button at the top right of your console if you didn't want to type text into the console directly. This is mainly recommended for longer commands.
+image:images/console-textbox.png[500,500]
+
+. Finally, enable the tang service in systemd and open the HTTP port in firewall:
++
+[source, text]
+[root@nbde1 ~]# systemctl enable tangd.socket --now
+[root@nbde1 ~]# firewall-cmd --add-service=http
+[root@nbde1 ~]# firewall-cmd --add-service=http --permanent
+
+Since tangd uses the systemd socket activation mechanism, the server starts as soon as the first connection comes in. A new set of cryptographic keys is automatically generated at the first start.
+
+== Lab 3.2 Installing the cockpit-storaged
+Your NBDE2 server is already encrypted and this exercise will install the client packages that will allow you to un-encrypt the root device drive upon boot without manual intervention.
+
+. Navigate to the Lab information page from *Lab 0 Setup steps*. This page has your environment's power control and consoles. Click on the console for *NBDE2*. The passphrase for the LUKs encrypted disk on *NBDE2* is *r3dh4t1!*. You will need to enter the pass phrase to complete the boot process.  If you wanted to copy and paste text to send to the console, press the *T* button at the top right of your console.
++
+image:images/lab3-console2.png[200,200]
+image:images/console-textbox.png[500,500]
+
+. Login to the NBDE2 server as *root* using the password *r3dh4t1!*.
+. Then, install the clevis packages:
++
```

Comment:
This mentions clevis but the command contains cockpit-storaged

Reply:
Fixed.